x/evm/types/params.go

```diff
@@ -153,10 +146,23 @@ func validateEIPs(i interface{}) error {
 		return fmt.Errorf("invalid EIP slice type: %T", i)
 	}
 
+	uniqueEIPs := make(map[int64]struct{})
+	duplicateEIPs := make([]int64, 0, len(eips))
+
 	for _, eip := range eips {
 		if !vm.ValidEip(int(eip)) {
 			return fmt.Errorf("EIP %d is not activateable, valid EIPS are: %s", eip, vm.ActivateableEips())
 		}
+
+		if _, ok := uniqueEIPs[eip]; ok {
+			duplicateEIPs = append(duplicateEIPs, eip)
```

Comment:
this should return error immediately

Reply:
why? Wouldn't it be better UX if all of the duplicates are printed? This way if there are multiple duplicates, all of them are listed and not just the first one.